script.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop:disable Style/CaseEquality, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
```

Comment:
* [ ] compare results in each of the examples below 
``` p [1,2,3].my_each_with_index{|a| a} ```
``` p [1,2,3].each_with_index{|a| a} ```

Reply:
method my_each_with_index updated